core/src/com/unciv/logic/automation/unit/UnitAutomation.kt

```diff
@@ -262,16 +263,63 @@ object UnitAutomation {
+        if (UnitActions.getUnitActions(unit).any { it.type == UnitActionType.HurryResearch }
+                && unit.name.hashCode() % 2 > 0) {
+            UnitActions.getUnitActions(unit)
+                .first { it.type == UnitActionType.HurryResearch }.action!!.invoke()
+            return
+        }
+
+        // Great merchant -> Conduct trade mission in about 50% of cases if not at war.
+        // TODO: This could be more complex to walk to the city state that is most beneficial to
+        //  also have more influence.
+        if (unit.name.hashCode() % 2 >= 0
+                && unit.hasUnique(UniqueType.CanTradeWithCityStateForGoldAndInfluence)
+                // Don't wander around with the great merchant when at war. Barbs might also be a
+                // problem, but hopefully by the time we have a great merchant, they're under
+                // control.
+                && !unit.civ.isAtWar()
+        ) {
+            val tradeMissionCanBeConductedEventually =
+                    SpecificUnitAutomation.conductTradeMission(unit)
+            if (tradeMissionCanBeConductedEventually)
+                return
+        }
+
+        // Great engineer -> Try to speed up wonder construction
+        if (unit.hasUnique(UniqueType.CanSpeedupConstruction) || unit.hasUnique(UniqueType.CanSpeedupWonderConstruction)) {
+            val wonderCanBeSpedUpEventually = SpecificUnitAutomation.speedupWonderConstruction(unit)
+            if (wonderCanBeSpedUpEventually)
+                return
+        }
+
+
+        // This has to come after the individual abilities for the great people that can also place
+        // instant improvements (e.g. great scientist).
         if (unit.hasUnique(UniqueType.ConstructImprovementConsumingUnit)
-                || unit.hasUnique(UniqueType.ConstructImprovementInstantly))
-        // catch great prophet for civs who can't found/enhance/spread religion
-            return SpecificUnitAutomation.automateImprovementPlacer(unit) // includes great people plus moddable units
+                || unit.hasUnique(UniqueType.ConstructImprovementInstantly)
+        ) {
+            // catch great prophet for civs who can't found/enhance/spread religion
+            // includes great people plus moddable units
+            val improvementCanBePlacedEventually =
+                    SpecificUnitAutomation.automateImprovementPlacer(unit)
+            if (!improvementCanBePlacedEventually)
+                tryStartGoldenAge(unit)
+        }
 
-        // ToDo: automation of great people skills (may speed up construction, provides a science boost, etc.)
+        // TODO: The AI tends to have a lot of great generals. Maybe there should be a cutoff
+        //  (depending on number of cities) and after that they should just be used to start golden
+        //  ages?
 
         return // The AI doesn't know how to handle unknown civilian units
     }
 
+    private fun tryStartGoldenAge(unit: MapUnit) {
```

Comment:
Do, or do not
There is no "try"
(This function can't fail)

Reply:
My good man
All the functions that return booleans, above, don't have the "try" prefix
And this one does
That should be reversed